### PR TITLE
Feature: Two-pane markdown editor and display for Collections pages

### PR DESCRIPTION
**Pull request by kwajiehao**
**Overview**
This PR implements the two-pane display (SimpleMDE markdown editor and HTML render) for pages in collections. As with PR[#6]( https://github.com/isomerpages/isomercms-frontend/pull/6), users will be able to edit content on the left pane and see it displayed in real-time on the right pane. 

Feature:
- Edit Collections pages and render HTML output in real-time
- Up-to-date left navigation bar (accurately reflects all pages in the collection)
- Current page is highlighted on the left navigation bar

### CODE REVIEW

File: src/templates/pageComponents/LeftNav.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          className="resize-sensor"
+          style={resizeStyle}
+        >
+          <div
+            className="resize-sensor-expand"
+            style={resizeStyle}
+          >
+            <div
+              style={{
+                position: 'absolute',
```
[the remark this answers is not in the document]
Comment (prestonlimlianjie) [reply src/templates/pageComponents/LeftNav.jsx]: After discussion, we decided to ignore the unnamed inline-styles for now; we'll come back to clean this up in the future.

Things to think about (maybe in a year): should we move the main Isomer template to Gatsby.js to make the template seamlessly translatable to the CMS?

File: src/layouts/EditCollectionPage.jsx [hunk 2]
```diff
@@ -94,44 +124,50 @@ export default class EditCollectionPage extends Component {
...
+  onEditorChange = (value) => {
+    this.setState({ editorValue: value });
+  }
+
   render() {
-    const { content, sha } = this.state;
-    const { match } = this.props;
+    const { match, location } = this.props;
     const { collectionName, fileName } = match.params;
+    const { pages } = location.state;
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply src/layouts/EditCollectionPage.jsx]: Set up an issue #19 to track this problem

File: src/templates/pageComponents/LeftNav.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      <aside className="bp-menu is-gt sidebar__inner" style={{ position: 'relative' }}>
+        <ul className="bp-menu-list">
+          {
+            pages.map((page) => {
+              return (
+                <li>
+                  {
+                    page.fileName === fileName ?
+                      <a className="is-active" href={page.path}>{deslugifyCollectionPage(page.fileName)}</a> :
+                      <a className="" href={page.path}>{deslugifyCollectionPage(page.fileName)}</a>
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply src/templates/pageComponents/LeftNav.jsx]: For this, I was trying to provide the correct URL to the page (although we have talked about disabling all links). In this case, `page.filename` provides the file name (such as `0-sub-page-A.md`) and `page.path` provides a string (such as `_left-nav-one%2F0-sub-page-A.md`) which we can manipulate into the proper URL. I've made the necessary adjustments to `page.path` to get the correct URL.

File: src/layouts/CollectionPages.jsx [hunk 4]
```diff
@@ -65,7 +65,15 @@ Pages in Collection
         {pages.length > 0
           ? pages.map((page) => (
             <li>
-              <Link to={`/sites/${siteName}/collections/${collectionName}/${page.fileName}`}>{page.fileName}</Link>
+              <Link to={{
+                pathname: `/sites/${siteName}/collections/${collectionName}/${page.fileName}`,
+                state: {
+                  pages,
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply src/layouts/CollectionPages.jsx]: Renamed to `leftNavPages`

File: src/layouts/EditCollectionPage.jsx [hunk 5]
```diff
@@ -52,8 +73,17 @@ export default class EditCollectionPage extends Component {
       const { match } = this.props;
       const { siteName, collectionName, fileName } = match.params;
       const { state } = this;
-      const base64Content = Base64.encode(this.contentBox.innerHTML);
-      const params = { content: base64Content, sha: state.sha };
+      const { editorValue, frontMatter } = state;
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply src/layouts/EditCollectionPage.jsx]: Done

**Pull request merged**